src/main/scala/de/htwg/se/sudoku/model/fileIoComponent/fileIoMongoDBImpl/FileIO.scala

```diff
@@ -0,0 +1,80 @@
+package de.htwg.se.sudoku.model.fileIoComponent.fileIoMongoDBImpl
+
+import com.google.inject.{Guice, Inject}
+import com.google.inject.name.{Named, Names}
+import de.htwg.se.sudoku.SudokuModule
+import de.htwg.se.sudoku.model.fileIoComponent.FileIOInterface
+import de.htwg.se.sudoku.model.gridComponent.GridInterface
+import net.codingwell.scalaguice.InjectorExtensions.ScalaInjector
+import org.mongodb.scala.model.Projections
+import org.mongodb.scala.{Document, MongoClient, MongoDatabase}
+import play.api.libs.json.{JsValue, Json}
+
+import scala.concurrent.Await
+import scala.concurrent.duration.Duration
+import scala.util.Try
+
+class FileIO @Inject()(@Named("MongoDBHost") host: String, @Named("MongoDBPort") port: Int) extends FileIOInterface {
+
+  private val mongoClient: MongoClient = MongoClient(s"mongodb://$host:$port")
+  private val db: MongoDatabase = mongoClient.getDatabase("sudoku-in-scala")
+  private val collection = db.getCollection("sudoku-in-scala-col")
+
+  override def load: Try[Option[GridInterface]] = {
+    val resF = collection.find().projection(Projections.excludeId()).toFuture()
+    val res = Await.result(resF, Duration.Inf)
+
+    var gridOption: Option[GridInterface] = None
+
+    Try {
+      val json: JsValue = Json.parse(res.head.toJson())
+      val size = (json \ "grid" \ "size").get.toString.toInt
+      val injector = Guice.createInjector(new SudokuModule)
```

Comment:
Why `SudokuModule` and not `MongoDBModule`?

Reply:
`MongoDBModule` only works if a mongo db database is installed, didn't want to make it the default

Reply:
The `src/main/scala/de/htwg/se/sudoku/model/fileIoComponent/fileIoMongoDBImpl/FileIO.scala ` is only used if [this](https://github.com/bb30/de.htwg.se.SudokuInScala/pull/14/files#diff-2f2697619d766493c74a61548e98ce43R26) line is inserted again or? If so we can use here the `MongoDBModule` and when we wan't to use it, we have to change only the other line not both...

Reply:
ah sorry, didn't catch that we weren't talking about the line you linked to